db/migrate/20191119111806_add_format_to_disk.rb

```diff
@@ -0,0 +1,5 @@
+class AddFormatToDisk < ActiveRecord::Migration[5.1]
+  def change
+    add_column :disks, :format, :string
```

Comment:
@borod108 format might be a little too general, what are some of the options for this value for RHV so we can get a feeling for how it might be used by other providers?

Reply:
Hi!
So format in this case is either qcow2 or raw.
We need this to validate in pre provision that we are trying to prevision something that fits the options described under: "Permitted Storage Combinations" here:
https://www.ovirt.org/documentation/admin-guide/chap-Virtual_Machine_Disks.html

Reply:
Okay, that sounds like the `:disk_type` column, on vmware that stores if it is thin, thick, or rdm.

Reply:
@agrare actually we use disk_type for that exactly.
These are more or less orthogonal for rhv - We can have "thin raw" disks on "file" type storage.

Reply:
Oh you're right I thought I checked the RHV parser and saw it wasn't used...I must have been looking at the wrong file

Reply:
I wonder if we should split up `:disk_type` into `:thin` and `:format` columns since currently it is a little overloaded.

```
          disk_type   |      thin      |     format
----------------------------------------------------
VMware |    thin      |      true      |     vddk
VMware |    thick     |     false      |     vddk
VMware | rdm-virtual  |     false      | virtual-rdm
VMware | rdm-physical |     false      | physical-rdm
Ovirt  |    thin      |     true       |     qcow
Ovirt  |    thick     |     false      |     qcow
Ovirt  |    thin      |     true       |     raw
Ovirt  |    thick     |     false      |     raw
```

Reply:
sounds good to me, but I just wonder if maybe we should have "sparsity" and in it "thin/preallocated" - just to keep it generic.

Reply:
Is preallocated the same thing as thick?  Trying to stay away from string columns because they tend to get less normalized aka different providers just put their own exact string there.

Reply:
@agrare I thought that is a good thing, anyway - yes thick is preallocated. so should the disk_type column just be deleted?

Reply:
Yes I think we should migrate disk_type into the new columns

Reply:
@agrare should the "thin" value be set to false for all the VMware formats other then "thin"?

Reply:
@agrare updated the PR, would you say this can work?

Reply:
@borod108 I think the logic is right we just need to try to make it more performant

Reply:
@agrare how about now?